README.md

```diff
@@ -54,6 +63,403 @@ npm install @dojo/shim
 
 The examples below are provided in TypeScript syntax. The package does work under JavaScript, but for clarity, the examples will only include one syntax.
 
+### Widget Routing
+
+Widgets are a fundamental concept for any Dojo 2 application and as such Dojo 2 Routing provides collection of components that integrate directly with existing widgets within an application.
+
+These components enable widgets to be registered against a route _without_ requiring any knowledge of the `Router` or `Routes`.
+
+#### Outlets
+
+The primary concept for the routing integration is an `outlet`, this is a unique identifier associated with registered application route. Dojo 2 Widgets can then be configured with these outlet identifiers using the `Outlet` higher order component. `Outlet` returns a "new" Widget that can be used like any other widget within a `render` method e.g. `w(MyFooOutlet, { })`.
+
+Properties can be passed to an `Outlet` widget, in the same way as if the original widget was being used, however all properties are made optional in order to allow the properties to be injected using the [mapParams](#mapParams) function described below.
+
+The number of widgets that can be mapped to a single outlet identifier is not restricted. All the configured widgets for a single outlet will be rendered when the route associated to the outlet is matched by the `router` and they are part of the current widget hierarchy.
+
+The following example configures a stateless widget with an outlet called `foo`, the resulting `FooOutlet` can be used in a widgets `render` in the same way as any other Dojo 2 Widget.
+
+```ts
+import { Outlet } from '@dojo/routing/Outlet';
+import { MyViewWidget } from './MyViewWidget';
+
+const FooOutlet = Outlet(MyViewWidget, 'foo');
+```
+
+Example usage of `FooOutlet`, where the widget will only be rendered when the route registered against outlet `foo` is matched.
+
+```ts
+class App extends WidgetBase {
+	protected render(): DNode {
+		return v('div', [
+			w(MyViewWidget, {})
+		]);
+	}
+}
+```
+
+#### Outlet Component Types
+
+When registering an outlet a different widget can be configure for each `MatchType` of a route:
+
+| Type              | Description |
+| ----------------- | ---------------------------------------------------- |
+|`MatchType.INDEX`  | This is an exact match for the registered route. E.g. Navigating to `foo/bar` with a registered route `foo/bar`.   |
+|`MatchType.PARTIAL`| Any match other than an exact match, for example `foo/bar` would partially match `foo/bar/qux` but only if `foo/bar/qux` was also a registered route. Otherwise it would be an `ERROR` match. |
+|`MatchType.ERROR`  | An error match occurs when a partial match occurs but there is no match for the next section of the route. |
```

Comment:
"An error match occurs when a partial match occurs but there is no match for the next section of the route." -> "When a partial match occurs but there is no match for the next section of the route."